lti_consumer/lti_1p3/consumer.py

```diff
@@ -625,6 +713,118 @@ def check_and_decode_deep_linking_token(self, token):
         # Return contentitems
         return content_items
 
+    def check_and_decode_proctoring_token(self, token):
+        """
+        Check and decode Proctoring LtiStartAssessment response and return relevant data to the Assessment Platform.
+        """
+        if not self.proctoring:
+            raise exceptions.LtiAdvantageServiceNotSetUp()
+
+        # Decode token and check expiration.
+        proctoring_response = self.tool_jwt.validate_and_decode(token)
+
+        # TODO: We MUST perform other forms of validation here.
+        # TODO: We MUST validate the verified_user claim if it is provided, although it is optional to provide it.
+        #       An Assessment Platform MAY reject the Start Assessment message if a required identity claim is missing
+        #       (indicating that it has not been verified by the Proctoring Tool).  Which identity claims a
+        #       Proctoring Tool will verify is subject to agreement outside of the scope of this specification but, at a
+        #       minimum, it is recommended that Proctoring Tools performing identity verification are able to verify the
+        #       given_name, family_name and name claims.
+        #       See 3.3 Transferring the Candidate Back to the Assessment Platform.
+
+        # Check Required LTI Claims
+        # -------------------------
+
+        # Check that the response message_type claim is "LtiStartAssessment".
```

Comment:
this stuff could probably be turned into a set of calls to a generic checker. They're mostly response.get(keystring) != None and then compare to some value so you could do it as check(response, keystring, desired value, errormessage)

but I'm not sure that would really be more readable or understandable